### PR TITLE
Keep Icon's subtree structure the same with and without a semanticLabel.

### PR DESCRIPTION
**Pull request by amirh**
This avoids rebuilding the subtree when a semanticLabel is set/unset.
Also updated some doc comments.

(This is a followup on post-merge comments for #12475)

### CODE REVIEW

File: packages/flutter/lib/src/widgets/icon.dart [hunk 1]
```diff
@@ -89,7 +89,10 @@ class Icon extends StatelessWidget {
   /// This would be read out in accessibility modes (e.g TalkBack/VoiceOver).
   /// This label does not show in the UI.
   ///
-  /// See [Semantics.label];
+  /// See also:
+  ///
+  ///  * [Semantics.label] which is set with [semanticLabel] in the underlying
+  ///  [Semantics] node.
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/icon.dart]: nit: indent this to align with [Semantics.label] in previous line.

Comment (goderbauer) [reply packages/flutter/lib/src/widgets/icon.dart]: Maybe `[Semantics] widget` to be a little more specific?

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/widgets/icon.dart (hunk 1 above)
Comment (amirh) [reply packages/flutter/lib/src/widgets/icon.dart]: done

File: packages/flutter/lib/src/widgets/icon.dart [hunk 2]
```diff
@@ -102,15 +105,19 @@ class Icon extends StatelessWidget {
     final double iconSize = size ?? iconTheme.size;
 
     if (icon == null)
-      return _wrapWithSemantics(new SizedBox(width: iconSize, height: iconSize));
+      return new Semantics(
+        label: semanticLabel,
+        child: new SizedBox(width: iconSize, height: iconSize)
+      );
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/icon.dart]: braces

Comment (amirh) [reply packages/flutter/lib/src/widgets/icon.dart]: Followed up in #12529

**Pull request merged**